### PR TITLE
Activity Progress Tracker insertion and status query done. 

### PR DESCRIPTION
**Pull request by sutara-aayush**
Along with it I have removed some unnecessarily used widgets and added some of the widgets in the files as required.

### CODE REVIEW

File: lib/db/dao/activity_progress_dao.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          activityId,
+          userId
+        ]);
+    if (maps.length > 0) {
+      return ActivityProgress.fromMap(maps.first);
+    }
+    return null;
+  }
+
+  Future<int> getActivityProgressScoreByTopicIdAndUserId(
```
Comment (talapadi) [inline lib/db/dao/activity_progress_dao.dart]: This is not a score....you can call it ActivityProgress

File: lib/screens/drawing_screen.dart [hunk 2]
```diff
@@ -33,6 +38,9 @@ class DrawingScreenState extends State<DrawingScreen> {
 
   void _initData() async {
     _activity = await ActivityRepo().getActivity(widget.activityId);
+    User _user = AppStateContainer.of(context).state.loggedInUser;
+    print(await ActivityProgressRepo()
+        .insertActivityProgress(_user.id, widget.topicId, widget.activityId));
```
Comment (talapadi) [inline lib/screens/drawing_screen.dart]: What will happen if the user attempts the same activity multiple times?

File: lib/db/entity/activity_progress.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class ActivityProgress {
+  static const table = 'activityProgress';
+  static const userIdCol = 'userId';
+  static const topicIdCol = 'topicId';
+  static const activityIdCol = 'activityId';
+
+  String userId;
+  String topicId;
+  String activityId;
```
Comment (talapadi) [inline lib/db/entity/activity_progress.dart]: Can you add a timestamp as well to track when the user attempted this activity?

File: lib/db/dao/activity_progress_dao.dart (hunk 1 above)
Comment (sutara-aayush) [reply lib/db/dao/activity_progress_dao.dart]: Done

File: lib/screens/drawing_screen.dart (hunk 2 above)
Comment (sutara-aayush) [reply lib/screens/drawing_screen.dart]: Nothing just it will check if not there then insert.

File: lib/db/entity/activity_progress.dart (hunk 3 above)
Comment (sutara-aayush) [reply lib/db/entity/activity_progress.dart]: Yeah, Sure, Done!

**Pull request merged**